pubsublite/producer.go

```diff
@@ -200,16 +189,25 @@ func (p *Producer) ProcessBatch(ctx context.Context, batch *model.Batch) error {
 			}
 		}
 		topic := p.cfg.TopicRouter(event)
-		producer, ok := p.producer[topic]
+		// creating a publisher is expensive and we can't do it lazily so we call Load+LoadOrStore
```

Comment:
Sorry, I wasn't clear. In this specific context (`sync.Map` access) we can't create a value lazily:

What I have in mind for lazily creating a value is something like
```
map.LoadOrStore(key, func() any { 
  // lazily create the value if missing
})
```

(java has something similar, it's optimized and does 1 lookup, creating the value lazily if missing)

So we need to use `Load`+`LoadOrStore` (accessing the map twice) otherwise a single `LoadOrStore(topic, newpublisher())` wuould create a publisher (unused) for each batch.